hub/core/storage/lru_cache.py

```diff
@@ -270,3 +270,31 @@ def _list_keys(self):
         for key in self.cache_storage:
             all_keys.add(key)
         return list(all_keys)
+
+    def __getstate__(self):
+        """Returns the state of the cache, for pickling"""
+
+        # flushes the cache before pickling
+        if not hasattr(self, "read_only") or not self.read_only:
+            self.flush()
+        return {
+            "next_storage": self.next_storage,
+            "cache_storage": self.cache_storage,
+            "cache_size": self.cache_size,
+        }
+
+    def __setstate__(self, state):
+        """Recreates a cache with the same configuration as the state.
+
+        PS: While restoring the cache, we reset its contents.
+        In case the cache storage was local/s3 and is still accessible when unpickled (if same machine/s3 creds present respectively), the earlier cache contents are no longer accessible.
```

Comment:
```suggestion
        Note: 
            While restoring the cache, we reset its contents.
            In case the cache storage was local/s3 and is still accessible when unpickled (if same machine/s3 creds present respectively), the earlier cache contents are no longer accessible.
```